libs/application-generic/src/usecases/tier-restrictions-validate/tier-restrictions-validate.usecase.ts

```diff
@@ -82,14 +83,37 @@ export class TierRestrictionsValidateUsecase {
     return [];
   }
 
-  private buildCronDeltaDeferDuration(
-    command: TierRestrictionsValidateCommand,
-  ): number | null {
-    const cronExpression = parseCronExpression(command.cron);
-    const firstTime = cronExpression.next().toDate();
-    const secondTime = cronExpression.next().toDate();
+  private isCronDeltaDeferDurationExceededTier(
```

Comment:
i tried to create a simple "smart solution" however on every approach it became very complicated very fast, therefore I created a more simple solution where we validate the next 50 iterations of the crown with an exit if we pass 2 years.